mesheryctl/pkg/utils/healthcheck.go

```diff
@@ -290,3 +290,138 @@ func WaitForPodRunning(c *meshkitkube.Client, desiredPod, namespace string, time
 
 	return pollForPodRunning(c, namespace, desiredPodName, time.Duration(timeout)*time.Second)
 }
+
+// Returns condition function to indicate that all the components are deleted.
+// A component in terminating state is not considered deleted
+func areMesheryComponentsDeleted(c *meshkitkube.Client, namespace, componentType string) wait.ConditionFunc {
+	return func() (bool, error) {
+		switch componentType {
+		case "pod":
+			{
+				pods, err := c.KubeClient.CoreV1().Pods(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(pods.Items) == 0, nil
+			}
+
+		case "replicaset":
+			{
+				replicasets, err := c.KubeClient.AppsV1().ReplicaSets(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(replicasets.Items) == 0, nil
+			}
+
+		case "rolebinding":
+			{
+				rolebindings, err := c.KubeClient.RbacV1().RoleBindings(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(rolebindings.Items) == 0, nil
+			}
+
+		case "role":
+			{
+				roles, err := c.KubeClient.RbacV1().Roles(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(roles.Items) == 0, nil
+			}
+
+		case "statefulset":
+			{
+				statefulsets, err := c.KubeClient.AppsV1().StatefulSets(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(statefulsets.Items) == 0, nil
+			}
+
+		case "service":
+			{
+				services, err := c.KubeClient.CoreV1().Services(namespace).List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					return false, err
+				}
+				return len(services.Items) == 0, nil
+			}
+		}
+
+		return true, nil
+	}
```

Comment:
Let's not do this with switch case and do it for all resources in one-go.

also not pods/replicasets exactly we should be checking for higher level concepts such as deployments/services/roles.

Reply:
The problem is, even when the deployment is deleted, the pods are still in terminating state and are not actually yet deleted. So if we are waiting for the resources to be deleted, then we might not want to return before everything is actually deleted

Reply:
Nice point! Understood we can keep it this way then.

Reply:
Piyush can you check the comment on the original issue by Lee, We can also not add the polling behaviour for resources at all if we don't need to (it doesn't break anything)